multi-value-user-attribute-mapper/src/main/java/de/intension/mapper/saml/MappedValueUserAttributeMapper.java

```diff
@@ -0,0 +1,172 @@
+        customConfigProperties.add(property);
+    }
+
+    private final String providerId      = PROVIDER_ID_PREFIX.concat(super.getId());
+    private final String typeAndCategory = TYPE_AND_CATEGORY_PREFIX.concat(super.getDisplayCategory());
+
+    /**
+     * Set mapped value as user attribute from context.
+     */
+    public static void setMappedValueAsUserAttribute(UserModel user, IdentityProviderMapperModel mapperModel,
+                                                     BrokeredIdentityContext context)
+    {
+        String attribute = mapperModel.getConfig().get(USER_ATTRIBUTE);
+        if (attribute != null) {
+            List<String> contextAttrValues = getUserAttribute(context, attribute);
+            List<String> userAttribute = user.getAttributes().get(attribute);
+            if (userAttribute != null && !CollectionUtil.collectionEquals(contextAttrValues, userAttribute)) {
+                user.setAttribute(attribute, contextAttrValues);
+            }
+        }
+    }
+
+    /**
+     * Set mapped value as user attribute inside identity context.
+     */
+    public static void setMappedValueInContext(IdentityProviderMapperModel mapperModel,
+                                               BrokeredIdentityContext context)
+    {
+        String attribute = mapperModel.getConfig().get(USER_ATTRIBUTE);
+        if (StringUtil.isNotBlank(attribute)) {
+            List<String> attributeValues = getUserAttribute(context, attribute);
+            if (!attributeValues.isEmpty()) {
+                Set<String> mappedValues = mapValues(mapperModel, attributeValues);
+                if (!CollectionUtil.collectionEquals(mappedValues, attributeValues)) {
+                    context.setUserAttribute(attribute, new ArrayList<>(mappedValues));
+                }
+            }
+        }
+    }
+
+    /**
+     * Map incoming value list based on a key/value mapping configuration.
+     */
+    private static Set<String> mapValues(IdentityProviderMapperModel mapperModel, List<String> attributeValues)
+    {
+        Set<String> mappedValues = new HashSet<>();
+        Map<String, String> configMap = mapperModel.getConfigMap(VALUE_MAPPINGS);
+        if (!configMap.isEmpty()) {
+            for (String attributeValue : attributeValues) {
+                translateAndAddValue(mapperModel, configMap, attributeValue, mappedValues);
+            }
+        }
+        return mappedValues;
+    }
+
+    /**
+     * Translate single incoming value and add it to the Set of translated values.
+     */
+    private static void translateAndAddValue(IdentityProviderMapperModel mapperModel, Map<String, String> configMap, String attributeValue,
+                                             Set<String> mappedValues)
+    {
+        for (Map.Entry<String, String> entry : configMap.entrySet()) {
+            String key = entry.getKey();
+            try {
+                if (key.equals(attributeValue)
```

Comment:
Könnte es passieren, dass key an dieser Stelle null ist?

Reply:
Ja, da fehlt noch eine Prüfung auf "NULL", da der Key tatsächlich in der GUI nicht gesetzt werden muss.